src/Components/ArticleCard/ArticleCard.js

```diff
@@ -1,31 +1,54 @@
-import React from 'react'
-import {Link} from 'react-router-dom'
-import './ArticleCard.css'
+import React from "react";
+import { Link } from "react-router-dom";
+import "./ArticleCard.css";
 import emptyHeart from "../../Images/empty-heart.svg";
 import filledHeart from "../../Images/filled-heart.svg";
-import { useSaved } from '../../Contexts/SavedContext';
+import { useSaved, useSavedDispatch } from "../../Contexts/SavedContext";
 
 export const ArticleCard = ({ multimedia, title }) => {
-    const {saveArticle, deleteArticle, savedArticles} = useSaved()
-
-    return(
-        <div className='article-card'>
-        <Link to={`/${title}`} key={title} id={title} className='single-article-link'>
-            <img className='article-image' src={multimedia} alt='article' aria-label={multimedia}/>
-            <h2 className='article-title'>{title}</h2>
-        </Link>
-        {!savedArticles.includes(title) && (
-            <button onClick={() => saveArticle(title)}>
-            <img src={emptyHeart} alt='save article' />
+  const { savedArticles } = useSaved();
+  const dispatch = useSavedDispatch();
+  return (
+    <div className="article-card">
+      <Link
+        to={`/${title}`}
+        key={title}
+        id={title}
+        className="single-article-link"
+      >
+        <img
+          className="article-image"
+          src={multimedia}
+          alt="article"
+          aria-label={multimedia}
+        />
+        <h2 className="article-title">{title}</h2>
+      </Link>
+      {!savedArticles.includes(title) && (
+        <button
+          onClick={() =>
+            dispatch({ type: "ADD_TO_SAVED", payload: { article: title } })
```

Comment:
this is doing the same thing as your `saveArticle` but we are now able to mock the dispatch so it can be tested. This is purely a change to make testing easier.